backend/open_webui/custom_functions/tools/annuaire_tool.py

```diff
@@ -0,0 +1,82 @@
+import requests
+import re
+from bs4 import BeautifulSoup as bs
+
+def convert_to_markdown(article):
```

Comment:
This html to markdown conversion is not preserving the structure of the original html document.  Is this important for further processing?